Week 4: 권시훈 과제 제출

kanai1 opened the pull request.

kanai1 committed:
Merge pull request #3 from KGU-C-Lab/master

Week 3: 권시훈 Pull Request

kanai1 committed:
Week 4: 권시훈 과제 제출

Feat: 로그인/회원가입 화면 만들기
- 로그인/ 회원가입/ 마이페이지 만들기
- 회원가입시에 각 입력에 제한두기
- 마이페이지에서 유저정보 출력하기

chattymin commented on 권시훈/Week 4/과제/signup/app/src/main/java/com/kanai1/signup/LoginActivity.kt:
```diff
@@ -0,0 +1,63 @@
+    private var resultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()){
+        userinfo = it.data?.getSerializableExtra("userInfo", userInfoClass::class.java)
+        if (userinfo != null) {
+            setID(userinfo!!.id)
+            setPassword(userinfo!!.password)
```
!! 연산은 가능한한 쓰지 않는 것이 좋습니다!!!
다른 방법을 사용해보는건 어떨까요?

chattymin commented on 권시훈/Week 4/과제/signup/app/src/main/java/com/kanai1/signup/LoginActivity.kt:
```diff
@@ -0,0 +1,63 @@
+        setContentView(binding.root)
+
+        binding.activityLoginTvSignup.setOnClickListener{
+            println("signupbtn Pushed")
```
지금은 실제 결과물을 내는게 아니기때문에 괜찮지만, 디버그용 코드들은 전부 지우는게 좋습니다!!

chattymin commented on 권시훈/Week 4/과제/signup/app/src/main/java/com/kanai1/signup/SignupActivity.kt:
```diff
@@ -0,0 +1,76 @@
+
+    fun getID():String? {
+        var id = binding.activitySignupEtID.text.toString()
+        return if(id.length in 6..10 && !id.contains(" ")) id else null
```
예외처리 야무지네요!

chattymin commented on 권시훈/Week 4/과제/signup/app/src/main/java/com/kanai1/signup/SignupActivity.kt:
```diff
@@ -0,0 +1,76 @@
+
+    fun getMBTI(): String? {
+        var MBTI = binding.activitySignupEtMBTI.text.toString()
+        return if(MBTI.length == 4 || MBTI.none { it !in 'A'..'Z' && it !in 'a'..'z' }) MBTI else null
```
이부분은 조건이 잘못된 것 같네요!! 한번 더 검토해보면 좋을 것 같아요

chattymin commented on 권시훈/Week 4/과제/signup/app/src/main/java/com/kanai1/signup/SignupActivity.kt:
```diff
@@ -0,0 +1,76 @@
+            var MBTI = getMBTI()
+            var AboutMe = getAboutMe()
+
+            if(id == null) showErrorToast("id는 6~8자의 영문 및 숫자를 이용해주세요")
+            else if(password == null) showErrorToast("password는 8~12자의 영문 및 숫자를 이용해주세요")
+            else if(name == null) showErrorToast("이름은 공백을 제외하고 1자 이상을 이용해주세요")
+            else if(MBTI == null) showErrorToast("MBTI는 영문 4글자를 이용해주세요")
+            else if(AboutMe == null) showErrorToast("About Me는 공백제외 1자 이상을 이용해주세요")
+            else{
+                var userinfo = userInfoClass(id, password, name, MBTI, AboutMe)
+                val loginIntent = Intent(this, LoginActivity::class.java).apply{
+                    putExtra("userInfo", userinfo)
+                }
+
+                setResult(RESULT_OK, loginIntent)
+                if(!isFinishing) finish()
+            }
```
처리 깔끔하고 좋네요!!
하지만, 이렇게 분기가 많을때는 when을 사용하면 가독성이 더 좋을 것 같아요 :)

chattymin commented on 권시훈/Week 4/과제/signup/app/src/main/java/com/kanai1/signup/userInfoClass.kt:
```diff
@@ -0,0 +1,11 @@
+
+import java.io.Serializable
+
+data class userInfoClass(
+    var id: String,
+    var password: String,
+    var nickname: String,
+    var MBTI: String,
+    var AboutMe: String
+): Serializable
```
Parcelable이 아닌 Serializable를 사용하셨네요!!
둘 다 좋은 방법입니다! 멋지네요 :)

chattymin reviewed (approved):
코드가 점점 더 완벽해지네요!! 이번주도 좋은 코드 잘 감상했습니다 :)

kanai1 committed:
Week 4: 권시훈 과제 피드백 수정

Feat: 피드백 사항 수정
- 디버깅 코드 지우기
- !!연산자 사용하지 말기
- MBTI유효성 검사코드 수정
- 많은 분기는 when으로 처리하기
